### PR TITLE
Add test coverage for revisereplica and retain in pkg/resourceinterpr…

### PR DESCRIPTION
**Pull request by NishantBansal2003**
**What type of PR is this?**
/kind feature
/kind failing-test

**What this PR does / why we need it**:
The test case coverage for pkg/resourceinterpreter/default/native/revisereplica has been increased from 66.67% to 100% and  pkg/resourceinterpreter/default/native/retain has been increased from 21.11% to more than 80%.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
To verify the changes in the pkg/resourceinterpreter/default/native/ directory run the following commands:
```
go test ./... -coverprofile=coverage.out
go tool cover -html=coverage.out -o coverage.html
open coverage.html
```
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
Hey @XiShanYongYe-Chang, PTAL

File: pkg/resourceinterpreter/default/native/retain_test.go [hunk 1]
```diff
@@ -39,6 +42,12 @@ func Test_retainK8sWorkloadReplicas(t *testing.T) {
 			Replicas: &observedNum,
 		},
 	})
+	observed2, _ := helper.ToUnstructured(&appsv1.Deployment{
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/default/native/retain_test.go:45]: How about renaming this variable to a meaningful word? The previous code may not be named well enough, can you help modify it by the way?

File: pkg/resourceinterpreter/default/native/retain_test.go [hunk 2]
```diff
@@ -256,3 +274,355 @@ func Test_retainPersistentVolumeClaimFields(t *testing.T) {
...
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{"name": "nginx"},
+				},
+			},
+			Replicas: &replicaNum,
+		},
+	})
+	observed2, _ := helper.ToUnstructured(&appsv1.Deployment{
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/default/native/retain_test.go:296]: `observed2` ditto

File: pkg/resourceinterpreter/default/native/retain_test.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/resourceinterpreter/default/native/retain_test.go:45]: The modified test code is more intuitive to read.

Comment (NishantBansal2003) [reply pkg/resourceinterpreter/default/native/retain_test.go:45]: Thanks for explaining so much clearly, Now updating it...

**Comment by NishantBansal2003** [comment]
@XiShanYongYe-Chang  Made some changes, PTAL...

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/approve

**Pull request merged**